main.go

```diff
@@ -384,6 +380,14 @@ func transformCompile(args []string) ([]string, error) {
 		return nil, fmt.Errorf("typecheck error: %v", err)
 	}
 
+	if envGarbleLiterals {
+		files = obfuscateLiterals(files, info)
+		pkg, err = origTypesConfig.Check(pkgPath, fset, files, info)
+		if err != nil {
+			return nil, fmt.Errorf("typecheck error: %v", err)
+		}
```

Comment:
hm, how come we didn't re-typecheck before, but now we do? you should document this.

it is a fairly large extra cost, but I assume it's OK since literal obfuscation is opt-in.

Reply:
Before wile didn't use any type info for string obfuscation.

Reply:
Ah, good point. I think you should still document that, since we modified the AST, we need to typecheck again.